### PR TITLE
cim 5.1 (new formula)

### PR DESCRIPTION
**Pull request by markjreed**
GNU implementation of the Simula programming language.

### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

### New Formulae Submissions:

- [X] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Have you built your formula locally prior to submission with `brew install <formula>`?

I get about 8 failures in 'brew tests', but no new ones after the addition of this formula.

### CODE REVIEW

File: Library/Formula/cim.rb [hunk 1]
```diff
@@ -0,0 +1,652 @@
+class Cim < Formula
+  desc "GNU Simula"
+  homepage "https://www.gnu.org/software/cim/"
+  url "http://ftpmirror.gnu.org/pub/cim/cim-5.1.tar.gz"
```
Comment (DomT4) [inline Library/Formula/cim.rb]: Please add a `https://ftp.gnu.org/gnu/` style mirror as well. You'll find lots of examples in other GNU formulae.

File: Library/Formula/cim.rb [hunk 2]
```diff
@@ -0,0 +1,652 @@
...
+  sha256 "b90717b66ec400503bdd69e537e8b7f3e8a9d106c3ba9a08a04ae57369a069a9"
+
+  patch :DATA
```
Comment (DomT4) [inline Library/Formula/cim.rb]: Please ensure this has been submitted upstream and include a link to that submission and a short explanation of what the patch is doing. Due to how extensive this patch is we're unlikely to accept this until it has been merged upstream.

File: Library/Formula/cim.rb [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    File.open("test.sim", "w") do |f|
```
Comment (DomT4) [inline Library/Formula/cim.rb]: You can use our `(testpath/"test.sim").append_lines` method here.

File: Library/Formula/cim.rb [hunk 4]
```diff
@@ -0,0 +1,652 @@
...
+      f.puts %{Begin\n  outtext("Hello, world.");\nEnd;}
+    end
+    system "#{bin}/cim", "--output=hello", "test.sim"
+    `./hello`.chomp == "Hello, world."
```
Comment (DomT4) [inline Library/Formula/cim.rb]: Use `assert_match "Hello, world.", shell_output("./hello")`

File: Library/Formula/cim.rb [hunk 5]
```diff
@@ -0,0 +1,659 @@
+class Cim < Formula
+  desc "GNU Simula"
+  homepage "https://www.gnu.org/software/cim/"
+  url "http://ftpmirror.gnu.org/cim/cim-5.1.tar.gz"
+  mirror "https://ftp.gnu.org/gnu/cim/cim-5.1.tar.gz"
+  sha256 "b90717b66ec400503bdd69e537e8b7f3e8a9d106c3ba9a08a04ae57369a069a9"
+
+  # patch submitted upstream, no response yet.
```
Comment (dunn) [inline Library/Formula/cim.rb]: Was it submitted via email, or to a bug tracker? If the former, add the submission date; if the latter, add the ticket URL in the comment, thanks!

**Comment by apjanke** [comment]
We've deprecated legacy-homebrew, and this PR needs to be squashed and rebased on `master` of the new homebrew/core.

Would you mind re-submitting this on the new [`homebrew/core` repo](https://github.com/homebrew/homebrew-core) repo?

**Comment by markjreed** [comment]
Will do.

On Tue, May 3, 2016 at 12:30 PM, Andrew Janke <notifications@github.com>
wrote:

> We've deprecated legacy-homebrew, and this PR needs to be squashed and
> rebased on master of the new homebrew/core.
>
> Would you mind re-submitting this on the new homebrew/core repo
> <https://github.com/homebrew/homebrew-core> repo?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/legacy-homebrew/pull/49788#issuecomment-216585840>
>



-- 
Mark J. Reed <markjreed@gmail.com>

**Pull request closed without merging**